uat/custom-components/client-java-sdk/src/main/java/com/aws/greengrass/testing/mqtt311/client/sdkmqtt/Mqtt311ConnectionImpl.java

```diff
@@ -153,7 +204,28 @@ public UnsubAckInfo unsubscribe(long timeout, final @NonNull List<String> filter
 
         stateCheck();
 
-        return null;
+        if (filters.size() != 1) {
+            throw new IllegalArgumentException("Iot device SDK MQTT v3.1.1 client does not support to unsubscribe from "
+                                            + "multiple filters at once");
+        }
+
+        final String filter = filters.get(0);
+
+        CompletableFuture<Integer> unsubscribeFuture = connection.unsubscribe(filter);
+        try {
+            Integer packetId = unsubscribeFuture.get(timeout, TimeUnit.SECONDS);
+            logger.atInfo().log("Unsubscribed on connection {} from topics filter {} packet Id {}", connectionId,
+                                    filter, packetId);
+            return new UnsubAckInfo(Collections.singletonList(0), null);
```

Comment:
for my understanding, why do we have a return code here?  looking at the mqtt 3.1.1 spec, unsuback doesn't have a payload / return code

Reply:
That kept compatibility of existing code of client, protocol, control and building blocks with MQTT  v5.0 which is the primary goal. All that code expecting reason code. 
Because we are going to implement 5 clients it is better to keep common code to be compatible with the most informative client. If some client does not provide such information we have two main options: 1) add hard-coded value to mandatory fields as did best client 2) kept optional fields not set.  We also have option 3) customize the common code like protocol, client's control and building blocks to adopt for each client. But that way code becomes rubbish very fast.